dgl_ptm/dgl_ptm/model/initialize_model.py

```diff
@@ -392,11 +394,27 @@ def step(self):
             #TODO add model dump here. Also check against previous save to avoid overwriting
             raise RuntimeError(f'execution of step failed for step {self.step_count}')
 
-    def run(self):
-        """ run the model for each step until the step_target is reached."""
+    def run(self, restart=False):
+        """
+        run the model for each step until the step_target is reached.
+
+        param: restart: boolean or int or a pair of ints, optional.
+        If True, the model is run from last checkpoint,
+        if an int, the model is run from the first milestone at that step,
+        if a pair of ints, the model is run from that milestone at that step.
```

Comment:
It seems that the first case is a special instance of the second case, where the first item is 1 i.e. restart =(1, int). can these two cases be merged to simplify the code?

Reply:
Based on our discussion, I will merge the int and tuple cases. The argument can then be a boolean (for running from a checkpoint) or a tuple (for running from a milestone).

Reply:
Thinking about it again, it is slightly unclear what to do with the second value of the tuple. We could allow setting this to None, but then the desired milestone to load might either be
* milestone 0 at the specified step count
* the most recent milestone at the specified step count. I'm leaving towards this option.

Reply:
To add some context: the use case where the milestone with step and increment would be important, is if we want to compare different policy choices that run until the same step.
Imagine we have some starting state (let's say at step 500) and we want to compare policy choices A, B, and C, after applying them for the same amount of time (let's say for 500 steps). So we'd restart the model 3 times from a milestone at step 500 (each with different config parameters representing the different policies) and run until step 1000 3 times and store the final state. In this case, we must be able to store multiple milestones at the same time step (1000).

Reply:
TODO: add logging message to state the milestone that will be used.
TODO: only keep the tuple case; remove the int case.